app/views/spree/layouts/_admin_body.html.haml

```diff
@@ -2,7 +2,7 @@
 = render "layouts/i18n_script"
 = yield :stripe_js
 
-#wrapper{ data: { hook: '' } }
+#wrapper{ data: { hook: '', controller: (yield :wrapper_data_controllers) } }
```

Comment:
Hmm yeah this isn't great either as it'll wrap all content on all admin pages. It might work for now but if we want to re-use this toggle controller on a page somewhere it could create unintended consequences. It's a tricky one to solve though.

One potential way of handling this kind of thing where different controllers or different instances of the same controller need to interact with each other across different parts of the DOM is to use a pub/sub approach using the [DOMEvents API](https://developer.mozilla.org/en-US/docs/Web/API/Event/Event). Basically we can just emit an event from one place and listen for that even tin another place.

I mentioned the code in the toggle method could be separated into two distinct functions, but maybe it's a new controller instead? So then if we think about what behaviour we're encapsulating and how it might be reusable; maybe this looks more like a "remote toggle" controller, which allows toggling some content in one part of the page from another part of the page?

I made quick Gist: https://gist.github.com/Matt-Yorkley/e907a125123f7d0574c0d396e19a395b

Does that make sense or does it look crazy? :sweat_smile:

Reply:
It's a fine line between engineering and over-engineering sometimes! If that last Gist is too much, we could swing the pendulum back in the opposite direction, and try the alternate approach in [this Gist](https://gist.github.com/Matt-Yorkley/b2fc378dcac735ade7e57f4eb9984a1d) instead. :sweat_smile:

Reply:
@Matt-Yorkley Good ideas and review thanks. The remote toggle controller looks better than adding that `controller: (yield :wrapper_data_controllers) }`. I have reworked this now using the approach in your second gist, that one is a bit easier/quicker to understand :sweat_smile: but I can change it to the first approach if needs be. I didn't rebase things yet.